### PR TITLE
str.extractall with no match returns appropriate MultIndex

### PR DESCRIPTION
**Pull request by jsnowacki**
Relates and fixes issue #19034. It's a small fix which creates empty MultiIndex instead of Index, with appropriate names, when `str.extractall` doesn't find any matches. Also, a related test in `pandas.tests.test_strings` has been updated to reflect the expected result.

### CODE REVIEW

File: pandas/tests/test_strings.py [hunk 1]
```diff
@@ -1074,26 +1074,28 @@ def test_extractall_single_group_with_quantifier(self):
 
     def test_extractall_no_matches(self):
```
Comment (jreback) [inline pandas/tests/test_strings.py]: can you parameterize this with both a 0-len index and a 0-len MI.

**Review by jreback — CHANGES_REQUESTED** [review]
tests comments & can you add a whatsnew for bug fixes

File: pandas/tests/test_strings.py [hunk 2]
```diff
@@ -1072,28 +1072,44 @@ def test_extractall_single_group_with_quantifier(self):
...
+        ([], (None, )),
+        ([], ('i1', )),
+        ([], (None, 'i2')),
+        ([], ('i1', 'i2')),
+        (['a3', 'b3', 'd4c2'], (None, )),
+        (['a3', 'b3', 'd4c2'], ('i1', 'i2')),
+        (['a3', 'b3', 'd4c2'], (None, 'i2')),
+        (['a3', 'b3', 'd4c2'], ('i1', 'i2')),
+    ])
+    def test_extractall_no_matches(self, data, names):
```
Comment (TomAugspurger) [inline pandas/tests/test_strings.py]: By test comments, @jreback meant a comment with the issue number, so like `# GH-19075` just below the `def test_`

**Review by TomAugspurger — APPROVED** [review]

**Comment by jsnowacki** [comment]
Ah, yes, sorry; didn't get this step. I've added it.

File: pandas/tests/test_strings.py [hunk 3]
```diff
@@ -1072,28 +1072,45 @@ def test_extractall_single_group_with_quantifier(self):
...
         r = s.str.extractall('(z)(z)')
-        e = DataFrame(columns=[0, 1])
+        e = DataFrame(columns=[0, 1], index=ei)
         tm.assert_frame_equal(r, e)
         # one named group.
         r = s.str.extractall('(?P<first>z)')
-        e = DataFrame(columns=["first"])
+        e = DataFrame(columns=["first"], index=ei)
         tm.assert_frame_equal(r, e)
         # two named groups.
```
Comment (jreback) [inline pandas/tests/test_strings.py]: can you put a blank line before each comment; easier to read the tests then

**Review by jreback — APPROVED** [review]
very minor comment. ping when pushed.

**Comment by jsnowacki** [comment]
@jreback I've pushed the comment newline fix.

**Comment by jreback** [comment]
thanks @jsnowacki

**Pull request merged**